packages/eslint-plugin/tests/rules/no-unsafe-unary-minus.test.ts

```diff
@@ -0,0 +1,31 @@
+import { RuleTester } from '@typescript-eslint/rule-tester';
+
+import rule from '../../src/rules/no-unsafe-unary-minus';
+import { getFixturesRootDir } from '../RuleTester';
+
+const rootDir = getFixturesRootDir();
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 2015,
+    tsconfigRootDir: rootDir,
+    project: './tsconfig.json',
+  },
+  parser: '@typescript-eslint/parser',
+});
+
+ruleTester.run('no-unsafe-unary-minus', rule, {
+  valid: [
+    '+42;',
+    '-42;',
+    '-42n;',
+    '(a: number) => -a;',
+    '(a: bigint) => -a;',
+    '(a: number | bigint) => -a;',
+    '(a: any) => -a;',
+    '(a: 1 | 2) => -a;',
+  ],
```

Comment:
[Testing] Some missing coverage for:

- A valid case that would be invalid with a different operator (e.g. `(a: string) => +a;`)
- Arrays
- Intersections
- Objects
- Strings
- Template literal strings
- `never`
- `unknown`
- `void`

Reply:
Done, thanks!